project/backend/src/test/java/com/google/sps/TestRestaurantParser.java

```diff
@@ -0,0 +1,45 @@
+package com.google.sps;
+
+import com.google.sps.data.JsonToRestaurantParser;
+import com.google.sps.data.Restaurant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TestRestaurantParser {
+
+  private static final int MISSING_FIELD = -1;
+  private static final List<String> EMPTY_TYPES = new ArrayList<>();
+  private static final String RESTAURANT_A = "RESTAURANT_A";
+  private static final String PLACE_ID_A = "A12345";
+  private static final String VICINITY_A = "10 Main Street";
+  private static final Map<String, Double> COORDS_A = Map.ofEntries(
+    Map.entry("lat", 10.0),
+    Map.entry("lng", 20.0)
+  );
+
+  @Test
+  public void noOptionalFields() throws JSONException {
+    JSONObject body = new JSONObject();
+    JSONObject geometry = new JSONObject();
+    geometry.put("location", COORDS_A);
+    body.put("place_id", PLACE_ID_A);
+    body.put("name", RESTAURANT_A);
+    body.put("vicinity", VICINITY_A);
+    body.put("types", EMPTY_TYPES);
+    body.put("geometry", geometry);
```

Comment:
so you can chain these put calls because the method returns the object back to you https://www.javadoc.io/doc/org.json/json/20170516/org/json/JSONObject.html#put-java.lang.String-java.lang.Object- (for the purposes of chaining) 

```
JSONObject geometry = new JSONObject().put(location, coordsA);
JSONObject body = new JSONObject().put(...).put(...).put(...); 
```